components/eamxx/src/control/intensive_observation_period.cpp

```diff
@@ -0,0 +1,329 @@
+#include "control/intensive_observation_period.hpp"
+
+#include "share/grid/point_grid.hpp"
+#include "share/io/scorpio_input.hpp"
+
+#include "ekat/ekat_assert.hpp"
+
+#include <pio.h>
+
+// Extend ekat mpi type for pair of double int, used
+// in find_closest_lat_lon_index_and_rank()
+namespace ekat {
+  template<>
+  MPI_Datatype get_mpi_type<std::pair<double, int>> () {
+    return MPI_DOUBLE_INT;
+  }
+}
+
+namespace scream {
+namespace control {
+
+// Helper functions for reading files
+namespace {
+// Currently EAMxx scorpio interface doesn't have a way to inquire about dimension names
+bool file_has_dim(const std::string& filename, const std::string& dimname)
+{
+  int ncid, dimid, err;
+
+  bool was_open = scorpio::is_file_open_c2f(filename.c_str(),-1);
+  if (not was_open) {
+    scorpio::register_file(filename,scorpio::FileMode::Read);
+  }
+
+  ncid = scorpio::get_file_ncid_c2f (filename.c_str());
+  err = PIOc_inq_dimid(ncid,dimname.c_str(),&dimid);
+  if (err==PIO_EBADDIM) return false;
+
+  EKAT_REQUIRE_MSG (err==PIO_NOERR,
+      "Error! Something went wrong while retrieving dimension id.\n"
+      " - filename : " + filename + "\n"
+      " - dimname  : " + dimname + "\n"
+      " - pio error: " + std::to_string(err) + "\n");
+  if (not was_open) {
+    scorpio::eam_pio_closefile(filename);
+  }
+  return true;
+}
+}
+
+IntensiveObservationPeriod::
+IntensiveObservationPeriod(const ekat::Comm& comm,
+	    	           const ekat::ParameterList& params)
+{
+  m_comm = comm;
+  m_params = params;
+  EKAT_REQUIRE_MSG(m_params.get<bool>("doubly_periodic_mode", false),
+                   "Error! Currently doubly_periodic_mode is the only use case for "
+	           "intensive observation period files.\n");
+
+  EKAT_REQUIRE_MSG(m_params.isParameter("target_latitude") && m_params.isParameter("target_longitude"),
+                   "Error! Using intensive observation period files requires "
+                   "target_lat and target_lon be gives as parameters in "
```

Comment:
I would spell out `lat` and `lon` in the message, since those are the names of the expected parameters.